File: backend/entity/game.go
package entity

import (
	"gorm.io/gorm"
)

type Game struct {
	gorm.Model
	GameName    string  `json:"game_name"`
	GamePrice   float64 `json:"game_price"` // ใช้ float64 ง่าย ๆ ถ้าอยากแม่นยำค่อยเปลี่ยนเป็น decimal lib
	Description string  `json:"description"`

	Threads   []Thread   `gorm:"foreignKey:GameID" json:"threads,omitempty"`
	UserGames []UserGame `gorm:"foreignKey:GameID" json:"user_games,omitempty"`
}

File: frontend/src/interfaces/Game.ts
// src/interfaces/Game.ts
export interface Game {
  ID: number;
  game_name: string;
  game_price: number;
  description: string;
  // relations
  threads?: Thread[];
  user_games?: UserGame[];
}

export interface CreateGameRequest {
  game_name: string;
  game_price: number;
  description: string;
}

export interface UpdateGameRequest {
  ID: number;
  game_name?: string;
  game_price?: number;
  description?: string;
}

import type { Thread } from "./Thread";
import type { UserGame } from "./UserGame";
